src/esp/sensor/CameraSensor.cpp

```diff
@@ -19,6 +19,17 @@ CameraSensor::CameraSensor(scene::SceneNode& cameraNode,
     : VisualSensor(cameraNode, spec),
       baseProjMatrix_(Magnum::Math::IdentityInit),
       zoomMatrix_(Magnum::Math::IdentityInit) {
+  // Sanity check
+  CORRADE_ASSERT(std::static_pointer_cast<VisualSensorSpec>(spec) != nullptr,
```

Comment:
Are you should static_pointer_cast would work? Usually such static_cast is used when you basically reverse an implicit conversion and it does NOT perform runtime checks. This is used when you are pretty sure you are going to refer to an object of a specific type, and thus a check is not necessary.

Please google static_pointer_cast, and dynamic_pointer_cast.

Reply:
Thanks, dynamic_pointer_cast seems like the way to go when checking types